apps/storefront/src/pages/OrderDetail/components/OrderBilling.tsx

```diff
@@ -21,6 +40,56 @@ export default function OrderBilling({ isCurrentCompany }: OrderBillingProps) {
 
   const b3Lang = useB3Lang();
 
+  const [isDigitalDownloadOpen, setIsDigitalDownloadOpen] = useState(false);
+  const [isDigitalProductLoading, setIsDigitalProductLoading] = useState(false);
+  const [digitalProducts, setDigitalProducts] = useState<OrderProductItem[]>([]);
+  const [currentDigitalProduct, setCurrentDigitalProduct] = useState<ProductItem | undefined>(
+    undefined,
+  );
+
+  const getDigitalProductsInformation = useCallback(async () => {
+    setIsDigitalProductLoading(true);
+    let elements = [];
+    try {
+      elements = orderId ? await getDigitalDownloadElements(orderId) : [];
+    } catch {
+      throw Error('Failed to fetch digital products');
+    } finally {
+      setIsDigitalProductLoading(false);
+    }
+
+    const digitalProductsData =
+      elements.length > 0 ? elements?.map((item: DigitalProduct) => item.node) : null;
+
+    const digitalProducts = billings[0].digitalProducts.map((product: OrderProductItem) => {
+      const fileUrls =
+        digitalProductsData?.find(
+          (item: DigitalProductNode) => item.productEntityId === product.product_id,
+        )?.downloadFileUrls || [];
+
+      return {
+        ...product,
+        downloadFileUrls: fileUrls,
+      };
+    });
+
+    setDigitalProducts(digitalProducts);
+  }, [billings, orderId]);
+
+  useEffect(() => {
+    if (orderId) {
+      getDigitalProductsInformation();
+    }
+  }, [orderId, getDigitalProductsInformation]);
+
+  const getCurrentDigitalProduct = (productId: number) => {
+    const current = digitalProducts.find(
+      (product: ProductItem) => product.product_id === productId,
```

Comment:
`digitalProducts` is already typed, no need to specify here:

```suggestion
      (product) => product.product_id === productId,
```